pkg/util/syncutil/map.go

```diff
@@ -0,0 +1,371 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// TODO(peter): This file was copied from sync/map.go in the Go stdlib. The
```

Comment:
Haven't reviewed in detail, but I believe you can use x/sync instead of adding this copy.